enumerable_methods.rb

```diff
@@ -0,0 +1,127 @@
+# rubocop: disable Metrics/ModuleLength
+# rubocop: disable Metrics/CyclomaticComplexity
+# rubocop: disable Metrics/PerceivedComplexity
+module Enumerable
+  def my_each
+    arr = self if self.class == Array
+    arr = flatten if self.class == Hash
+    arr = to_a if self.class == Range
+
+    count = 0
+    while count < arr.length
+      yield(arr[count], count)
+      count += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    my_each { |elem, count| yield(elem, count) }
+  end
+
+  def my_select
+    selected = []
+    my_each { |elem| selected << elem if yield(elem) }
+    selected
+  end
+
+  def my_all(arg = nil)
+    if block_given?
+      my_each { |elem| return false if yield(elem) != true }
+      return true
+    end
+
+    if arg.nil?
+      my_each { |elem| return false if elem.nil? || elem == false }
+    elsif arg.is_a? Class
+      my_each { |elem| return false if elem.class != arg }
+    elsif arg.class == Regexp
+      my_each { |elem| return false unless arg.match(elem) }
+    else
+      my_each { |elem| return false if elem != arg }
+    end
+    true
+  end
+
+  def my_any(arg = nil)
+    if block_given?
+      my_each { |elem| return true if yield(elem) == true }
+      return false
+    end
+
+    if arg.nil?
+      my_each { |elem| return true if elem.nil? || elem == true }
+    elsif arg.is_a? Class
+      my_each { |elem| return true if elem.class == arg }
+    elsif arg.class == Regexp
+      my_each { |elem| return true if arg.match(elem) }
+    else
+      my_each { |elem| return true if elem == arg }
+    end
+    false
+  end
+
+  def my_none(arg = nil)
+    arr = self
+    if block_given?
+      !arr.my_any { |elem| yield(elem) }
+    else
+      !arr.my_any(arg)
+    end
+  end
```

Comment:
```rb
2.6.5 :045 > false_array = [nil, false, nil, false]
2.6.5 :046 > false_array.none?
 => true 
2.6.5 :047 > false_array.my_none
 => false 
2.6.5 :048 >
```

Reply:
`my_none` is also returning `NoMethodError` as in the screenshot below, instead of returning `true`
![my_none](https://user-images.githubusercontent.com/46388113/89060612-d4408a80-d352-11ea-936c-1c9a8cd19a4c.png)